woocommerce/woocommerce-android
# Issue/12365 custom fields list

Closes: #12365 


### Description
This PR introduces a new screen to show the list of custom fields (metadata) of a given item, reusing the existing one for orders would require too much changes, and as I'm aiming for a reactive approach, I decided to create a new one instead.

The PR got bigger than I anticipated, but I think the code is quite simple to review, if you prefer to have it split, please let me know and I can attempt to do so.

### Testing information
1. Open an order in wp-admin.
2. Confirm the "Custom Fields" section is visible (from the screen options at the top of the page)
3. Add some custom fields (for a more comprehensive testing, add fields that contain a URL, an email and a phone number)
4. Open the order in the app.
5. Tap on "View custom fields"
6. Confirm the fields are loaded.
7. Pull the list and confirm this refreshes the list of fields.
8. Confirm the URL, Email, and Phone values are clickable and test clicking on them.

### The tests that have been performed
The above steps in addition to testing custom fields that contain HTML

### Images/gif
<img src="https://github.com/user-attachments/assets/9bb8c3a8-3398-4159-8a8f-26de7972e214" width=360 />


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
